nilearn/input_data/base_masker.py

```diff
@@ -5,75 +5,85 @@
 # License: simplified BSD
 
 import warnings
+import abc
 
 import numpy as np
-import itertools
 
 from sklearn.base import BaseEstimator, TransformerMixin
-from sklearn.externals.joblib import Memory, Parallel, delayed
+from sklearn.externals.joblib import Memory
 
 from .. import masking
 from .. import image
 from .. import signal
 from .. import _utils
 from .._utils.cache_mixin import CacheMixin, cache
-from .._utils.class_inspect import enclosing_scope_name, get_params
-from .._utils.compat import _basestring, izip
-from nilearn._utils.niimg_conversions import (
-    _iter_check_niimg, _check_same_fov)
-
-
-def filter_and_mask(imgs, mask_img_,
-                    parameters,
-                    memory_level=0,
-                    memory=Memory(cachedir=None),
-                    verbose=0,
-                    confounds=None,
-                    copy=True,
-                    sample_mask=None):
+from .._utils.class_inspect import enclosing_scope_name
+from .._utils.compat import _basestring
+
+
+def filter_and_extract(imgs, extraction_function,
+                       parameters,
+                       memory_level=0, memory=Memory(cachedir=None),
+                       verbose=0,
+                       confounds=None,
+                       copy=True):
+    """Extract representative time series using given function.
+
+    Parameters
+    ----------
+    imgs: images
+        Images to be masked
+
+    extraction_function: function
+        Function used to extract the time series from 4D data.
+
+    For all other parameters refer to NiftiMasker documentation
+    """
+    # Since the calling class can be any *Nifti*Masker, we look for exact type
+    if verbose > 0:
+        class_name = enclosing_scope_name(stack_level=3)
```

Comment:
I am not sure how enclosing_scope_name works but it feels like the magical stack_level set to 3 may not work for all cases since you don't have any idea how far away in the stack the class is.